demos/codelab/app/scripts/music_maker.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright 2017 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ const MusicMaker = {
+  queue_: [],
+  player_: new Audio(),
+  queueSound: function(soundUrl) {
+    this.queue_.push(soundUrl);
+  },
+  play: function() {
+    let next = this.queue_.shift();
+    if (next) {
+      this.player_.src = next;
+      this.player_.play();
```

Comment:
This sound implementation is great in its simplicity at the code level, but will behave very differently than the iOS app because only one sound can play at time.

That said, the other half of the simultaneity problem is in running multiple event handler scripts. I hope to resolve that through the new JS Interpreter features.

It would be a huge help to solve the above problem before-hand. Not necessarily this PR. The Web Audio API is a better approach than `<audio>` tags / `Audio` object. [This tutorial](https://www.html5rocks.com/en/tutorials/webaudio/intro/) covers everything we would need out of `AudioContext` and `AudioNodes` (and a little more).

Reply:
I got it. I submitted an issue: https://github.com/google/blockly/issues/1330 Feel free to assign me (I can't do it myself).